src/plans/nonmutating_manifolds_plans.jl

```diff
@@ -14,31 +14,58 @@ end
 function linesearch_backtrack(
     M::NONMUTATINGMANIFOLDS,
     F::TF,
-    x,
+    p,
     gradF::T,
     s,
     decrease,
     contract,
     retr::AbstractRetractionMethod=ExponentialRetraction(),
     η::T=-gradF,
-    f0=F(x);
-    stop_step=0.0,
+    f0=F(p);
+    stop_when_stepsize_less=0.0,
+    stop_when_stepsize_larger=max_stepsize(M, p),
+    max_increase_steps=100,
+    max_decrease_steps=1000,
 ) where {TF,T}
-    x_new = retract(M, x, s * η, retr)
-    fNew = F(x_new)
-    while fNew < f0 + decrease * s * real(inner(M, x, η, gradF)) # increase
-        x_new = retract(M, x, s * η, retr)
-        fNew = F(x_new)
+    msg = ""
+    p_new = retract(M, p, s * η, retr)
+    fNew = F(p_new)
+    i = 0
+    while fNew < f0 + decrease * s * real(inner(M, p, η, gradF)) # increase
+        i = i + 1
         s = s / contract
+        p_new = retract(M, p, η, s, retr)
+        fNew = F(p_new)
+        if i == max_increase_steps
+            msg = "Max increase steps ($(max_increase_steps) reached"
+            break
+        end
+        if s > stop_when_stepsize_larger
```

Comment:
Be careful -- `max_stepsize` returns the maximum length of the vector we can go in one step, so I think this should be something like `s * norm(M, p, η) > stop_when_stepsize_larger`

Reply:
well-spotted. Will fix that.

Reply:
Since the parameter is called `when step_size_larger` I will prefer to scale the default number of said parameter instead

Reply:
Sure, that's also fine.